client_connection/manager_test.go

```diff
@@ -18,9 +18,10 @@ import (
 
 type test_context struct {
 	suite.Suite
-	connManager         *connectionManager
-	fakeDiscoveryClient *server.ClientFake
-	fakeOpenVpn         *fake_openvpn_client
+	connManager                *connectionManager
+	fakeDiscoveryClient        *server.ClientFake
+	fakeOpenVpn                *fake_openvpn_client
+	fakeDialogResumeDisconnect chan int
```

Comment:
Why can't we move this channel to a field inside fake_dialog struct ?
And define two specials operations on fake_dialog itself ?

Reply:
because fake_dialog is not accessible from this context and is private to connection manager.